### DISCUSSION TITLE
Blazorize Material DataGrid: Vertical grid lines are not visible. And also tell me how to left align DataGrid Caption/Column Header

### DISCUSSION THREAD

**Discussion by shakeeluetian** [Megabit/Blazorise#2764] [General]
a)  How to show `DataGrid `vertical Bordered/Grid lines although I have made `Bordered="true"`
`    <DataGrid  TItem="CompanyModel" Class="my-data-grid" Data="@companies" ShowPager="true" Editable="true" EditMode="DataGridEditMode.Popup" Striped="true" FixedHeader="false" CustomFilter="@OnCustomFilter" Style="font-size: 0.85rem;padding:0px;" Bordered="true" Responsive="true" Narrow="true" PageSize="10">`

b) How to align DataGrid Column Heading/Caption
i have used `HeaderTextAlignment="TextAlignment.Left"` in `<DataGridColumn>` but not aligned

**Comment by stsrki** [comment]
That is the expected behavior for tables in both Material and Bootstrap. When `Responsive="true"` (`class="table-responsive"`) is applied it will remove vertical borders. My guess is it's because in the responsive mode you would scroll horizontally and it should look weird with the borders. You can see the exact same appearance on their own documentation:

- https://getbootstrap.com/docs/4.6/content/tables/#responsive-tables
- https://djibe.github.io/material/docs/4.6/content/tables/#responsive-tables

**Reply by shakeeluetian** [reply]
![aaaaaa](https://user-images.githubusercontent.com/88061542/127920961-0e630984-a45b-44c1-9b39-eaf91d4b394c.JPG)

**Reply by shakeeluetian** [reply]
I have used `<DataGridColumn HeaderTextAlignment="TextAlignment.Left" `  but Caption/Header is not exactly aligned. Why??

**Reply by stsrki** [reply]
HeaderTextAlignment should work. I just tried it.

![image](https://user-images.githubusercontent.com/900302/127921357-90ef2054-ca72-4e49-b9b1-7204159d2690.png)

---

Can you inspect your table and see what is applied in DevTools?

Also, what Blazorise version do you use?

**Reply by shakeeluetian** [reply]
![aaa](https://user-images.githubusercontent.com/88061542/127921801-03032e56-bb2e-4284-9744-ae885d57d25e.JPG)

**Reply by shakeeluetian** [reply]
I am using  DataGrid and for grid lines CSS is ` border: 0.01px solid lightgray'
`

**Reply by shakeeluetian** [reply]
![aaaaaaaaaaaaaaa](https://user-images.githubusercontent.com/88061542/127922944-7e2e9eb9-8a27-4f20-b338-55119d3128eb.JPG)

But then why Header is not Left aligned?? 
`<DataGridColumn HeaderTextAlignment="TextAlignment.Left" Width="23%"   TItem="CompanyModel" Field="@nameof(CompanyModel.CompanyID)"  Caption="Company" Editable="true" Sortable="true" />
`

**Reply by shakeeluetian** [reply]
.my-data-grid tr td {
        **padding: 0.1rem;**                               **//This was the culprit. Thanks**
        border: 0.01px solid lightgray;
    }

**Reply by shakeeluetian** [reply]
Thanks a lot @stsrki  for prompt response
